File: backend/app/services/auth_service.py
import re

from werkzeug.security import check_password_hash, generate_password_hash


def hash_password(password: str) -> str:
    return generate_password_hash(password)


def verify_password(password_hash: str, password: str) -> bool:
    return check_password_hash(password_hash, password)


def normalize_phone(value: str) -> str:
    raw = (value or "").strip()
    if not raw:
        raise ValueError("Contact number is required")
    digits = re.sub(r"\D", "", raw)
    if len(digits) < 7 or len(digits) > 15:
        raise ValueError("Contact number must be 7–15 digits (include country code)")
    return f"+{digits}"


def validate_password(password: str) -> None:
    if len(password) < 8:
        raise ValueError("Password must be at least 8 characters")

File: server/src/services/authService.ts
import { hashPassword as werkzeugHash, verifyWerkzeugPassword } from './werkzeugPassword.js'

export function hashPassword(password: string): string {
  return werkzeugHash(password)
}

export function verifyPassword(passwordHash: string, password: string): boolean {
  return verifyWerkzeugPassword(passwordHash, password)
}

export function normalizePhone(value: string): string {
  const raw = (value ?? '').trim()
  if (!raw) throw new Error('Contact number is required')
  const digits = raw.replace(/\D/g, '')
  if (digits.length < 7 || digits.length > 15) {
    throw new Error('Contact number must be 7–15 digits (include country code)')
  }
  return `+${digits}`
}

export function validatePassword(password: string): void {
  if (password.length < 8) {
    throw new Error('Password must be at least 8 characters')
  }
}
